opengovsg/postmangovsg
# feat: support whitelisting domains through `agencies` table

## Problem

Closes #1115 

## Solution

**Features**:

- Add `Agency` model and table in database
- Validate whitelisted domain using both config domains and agencies table domains

**Improvements**:
- Log matched domain for each authentication request

## Tests
- Add email domain (e.g. `@gmail.com`) to agencies table using the following query:
```sql
INSERT INTO agencies VALUES('@gmail.com', CURRENT_TIMESTAMP, CURRENT_TIMESTAMP);
```
- Check that you will be able to request for the OTP with an email belonging to the newly inserted domain
- Delete the domain in `agencies` table
- Check that you will not be able to request for an OTP with an email belonging to the deleted domain
